examples/dtls-echo/dtls-client.c

```diff
@@ -102,18 +104,18 @@ static void dtls_handle_read(dtls_context_t *ctx)
 
     if (sock_udp_get_remote(sock, &remote) == -ENOTCONN) {
         DEBUG("%s: Unable to retrieve remote!\n", __func__);
-        return;
+        return 0;
     }
 
     ssize_t res = sock_udp_recv(sock, packet_rcvd, DTLS_MAX_BUF,
                                 1 * US_PER_SEC + DEFAULT_US_DELAY,
                                 &remote);
 
-    if (res <= 0) {
+    if (res < 0) {
```

Comment:
Does it make sense to continue if `res == 0`, that is

> no received data is available, but everything is in order. <sup>[`sock_udp_recv()`](https://riot-os.org/api/group__net__sock__udp.html#gad1030df9b925878c2bf678487a9c88f7)</sup>

Reply:
In this case yes, because DTLS alerts have payload size 0 but are important for the client/server to know if they received any alerts.

Reply:
So that is UDP payload size? The signalling happens by sending an empty message to the DTLS port?

Reply:
> So just to be clear, that is UDP payload size? 

Yepp, like with normal udp sockets;

Reply:
wait. now that you mention it, I might be wrong here. I'll check again.

Reply:
So, I somehow had the wrong idea in mind that a DTLS alert has payload size 0 eventhough I already saw many times in wireshark that it isn't.



> In this case yes, because DTLS alerts have payload size 0 but are important for the client/server to know if they received any alerts.

This statement is wrong.

> The signalling happens by sending an empty message to the DTLS port?

No, the signalling happens by sending a DTLS message with type alert in a DTLS record header, which means the received payload size cannot be 0.

Receiving a UDP payload size 0 is technically not an error *but* it doesn't make sense to continue if  `res == 0` because we are expecting a DTLS packet and not just UDP. So I'll revert this change and check if I made the same mistake at other places.

Sorry for the confusion :)